src/v/datalake/coordinator/coordinator.cc

```diff
@@ -146,4 +293,20 @@ coordinator::sync_get_last_added_offset(model::topic_partition tp) {
     co_return prt_state.pending_entries.back().last_offset;
 }
 
+void coordinator::notify_leadership(std::optional<model::node_id> leader_id) {
+    auto node_id = stm_->raft()->self().id();
+    bool is_leader = leader_id && *leader_id == stm_->raft()->self().id();
```

Comment:
nit: I think an optional equality handles disengaged state too (don't have to explicitly check if its set).

eg: (nullopt == node_id{1}) == false.

Reply:
ack, will do in a follow up